### ISSUE TITLE
Possible bug in activate_helper.py (source_foreign) - stack trace included

### ISSUE THREAD

**Issue by bytesemantics** [Cadair/jupyter_environment_kernels#25]
Hi

I've added mulitple virtual env paths to my .condarc file (envs_dirs property).
These paths are in Windows file URI format and of the form \machine\dir1\dir2\dir3\venvs 

I get the following error stack:

```
[E 09:01:07.515 NotebookApp] Couldn't get environment variables for commands: ['activate', '\\\\machine\\di1\\dir2\\dir3\\venvs\\myEnv']
    Traceback (most recent call last):
      File "C:\Users\43903265\AppData\Local\Continuum\Anaconda3\lib\site-packages\environment_kernels-1.0-py3.5.egg\environment_kernels\envs_conda.py", line 48, in _get_env_vars_for_conda_env
        return source_env_vars_from_command(args)
      File "C:\Users\43903265\AppData\Local\Continuum\Anaconda3\lib\site-packages\environment_kernels-1.0-py3.5.egg\environment_kernels\activate_helper.py", line 57, in source_env_vars_from_command
        return source_cmd(args)
      File "C:\Users\43903265\AppData\Local\Continuum\Anaconda3\lib\site-packages\environment_kernels-1.0-py3.5.egg\environment_kernels\activate_helper.py", line 105, in source_cmd
        return source_foreign(args, stdin=stdin)
      File "C:\Users\43903265\AppData\Local\Continuum\Anaconda3\lib\site-packages\environment_kernels-1.0-py3.5.egg\environment_kernels\activate_helper.py", line 184, in source_foreign
        seterrpostcmd=ns.seterrpostcmd)
      File "C:\Users\43903265\AppData\Local\Continuum\Anaconda3\lib\site-packages\environment_kernels-1.0-py3.5.egg\environment_kernels\activate_helper.py", line 426, in foreign_shell_data
        universal_newlines=True)
      File "C:\Users\43903265\AppData\Local\Continuum\Anaconda3\lib\subprocess.py", line 626, in check_output
        **kwargs).stdout
      File "C:\Users\43903265\AppData\Local\Continuum\Anaconda3\lib\subprocess.py", line 708, in run
        output=stdout, stderr=stderr)
    subprocess.CalledProcessError: Command '['cmd', '/C', 'C:\\Users\\43903265\\AppData\\Local\\Temp\\tmp8you73_p.bat']' returned non-zero exit status 1
```

I have commented out line 434 "os.remove(tmpfile.name)" from activate_helper.py (def foreign_shell_data) - so I could determine what script is being run and its commands.

I find this in the generated batch file:
`call ^"C:\Users\XXXXX\AppData\Local\Continuum\Anaconda3\Scripts\activate^" ^"\\\machine\dir1\dir2\dir3\venvs\myEnv^"`

I guess there's a regexp bug  (note the 3 \\ at the beginning of the virtualenv URI).

**Comment by Cadair** [comment]
I am going to have to let @janschulz have a look at this, this code was his magic.

**Comment by janschulz** [comment]
Actually that looks good on first sight, IMO...

What happens when you execute `"C:\Users\XXXXX\AppData\Local\Continuum\Anaconda3\Scripts\activate" "\\machine\dir1\dir2\dir3\venvs\myEnv"`in cmd?

**Comment by bytesemantics** [comment]
@janschulz Hi - unfortunately I didn't pay too much attention to how my typing rendered....

The env URI has 3 '\' characters rather than 2.

I modified the argvquote function - and modified the else block at line 153.
this now reads:

else:
    if cmdline != '"\\':
        n_backslashes += 1


Now this works for me (in that it generates a valid URL) - but I've not run any of the available tests or considered anyone elses use cases.

If I run the generated bat file manually - it runs fine for me - the environment variables stack is pushed to console output - and it looks good.

However running as part of the environment_kernel_manager /jupyter integration - I still get a subprocess exception - which I narrowed to a CalledProcessError exception being thrown.
Printing out the exception details only tells me that the call returned an errorlevel of 1.

I've not got any further in the time allocated to spend on this - and would appreciate any thoughts/advice.
